packages/sit-onyx/src/components/OnyxPagination/OnyxInlinePagination.vue

```diff
@@ -0,0 +1,207 @@
+<script lang="ts" setup>
+import { iconChevronLeftSmall, iconChevronRightSmall } from "@sit-onyx/icons";
+import { computed } from "vue";
+import { useDensity } from "../../composables/density.js";
+import {
+  SKELETON_INJECTED_SYMBOL,
+  useSkeletonContext,
+} from "../../composables/useSkeletonState.js";
+import { injectI18n } from "../../i18n/index.js";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import OnyxSkeleton from "../OnyxSkeleton/OnyxSkeleton.vue";
+import OnyxVisuallyHidden from "../OnyxVisuallyHidden/OnyxVisuallyHidden.vue";
+import type { OnyxPaginationProps } from "./types.js";
+
+const props = withDefaults(defineProps<OnyxPaginationProps>(), {
+  skeleton: SKELETON_INJECTED_SYMBOL,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the selected page changes.
+   */
+  "update:modelValue": [page: number];
+}>();
+
+const { t } = injectI18n();
+const { densityClass } = useDensity(props);
+const skeleton = useSkeletonContext(props);
+
+const hasReachedMin = computed(() => props.modelValue <= 1);
+const hasReachedMax = computed(() => props.modelValue >= props.pages);
+
+const displayPagesNumbers = computed<Array<number | string>>(() => {
+  const currentPage = props.modelValue;
+  const totalPages = props.pages;
+
+  const pagesToShow = new Set<number>();
+
+  pagesToShow.add(1);
+  pagesToShow.add(totalPages);
+
+  pagesToShow.add(currentPage);
+  pagesToShow.add(currentPage - 1);
+  pagesToShow.add(currentPage + 1);
+
+  if (currentPage === 1) pagesToShow.add(3);
+  if (currentPage === totalPages) pagesToShow.add(totalPages - 2);
+
+  const uniqueSortedPages = Array.from(pagesToShow)
+    .filter((page) => page >= 1 && page <= totalPages)
+    .sort((a, b) => a - b);
+
+  const finalPages: Array<number | string> = [];
```

Comment:
```suggestion
  const finalPages: (number | "ellipsis")[] = [];
```

We can type this more strictly so it does only allow "ellipsis" and not all strings :) Also you can remove the generic type from the computed here so we don't have to type it two times. TypeScript will then just infer the type from the return value correctly